### ISSUE TITLE
invoke and github actions: migrate everything to use tox environments

### ISSUE THREAD

**Issue by MartinBernstorff** [MartinBernstorff/swift-python-cookiecutter#125]
Avoids issues like this one, where we have a dependency that's required for pyright and:

* Installed locally, hence `inv pr` succeeds
* Not installed remotely, hence `pyright .` fails

https://github.com/Aarhus-Psychiatry-Research/timeseriesflattener/actions/runs/4743483039/jobs/8423090479?pr=183

If all the commands use tox environments, we have complete parity.

Do you agree with this prioritisation, @KennethEnevoldsen?

**Comment by MartinBernstorff** [comment]
My ideal situation is:
`invoke > tox > utility`, where the utility is e.g. `docs, pyright, tests`.

This would mean that github actions only need to manage:
1. Setup python
2. Setup invoke and tox
3. Cache `.tox`
4. Run command

Essentially moving all the complexity to something that runs locally. Specifically, the environments the tasks run in locally and remotely will be exactly the same (except for, potentially, OS), making for way faster debugging.

It also allows for easy multi-python test-running. I tried it yesterday on timeseriesflattener; had some issues with python3.8, and using tox for running that environment was an awesome experience.

Yeah, I looked into `nox` yesterday, but seems their dependency parsing/environment caching is worse. It re-created the entire environment every time I re-run tasks. Tox only recreates if relevant dependencies in the toml have changed.
